### ISSUE TITLE
awsmobile publish should not redeploy the back end if it has not changed

### ISSUE THREAD

**Issue by baharev** [aws/awsmobile-cli#34]
When I run `awsmobile publish` the entire back end code is zipped, uploaded to S3, and redeployed, even though it has not changed. It is very annoying if I fix a small issue in the front end, and then I have to wait for several minutes for the otherwise unchanged backend to be redeployed.

Is there a way to avoid this and to iterate faster?

**Comment by UnleashedMind** [comment]
awsmobile-cli has this smarts coded in since version 1.0.4 
Please make sure to get the latest version (1.0.7) and try again. 
Please let us know if you still get this behavior.

**Comment by baharev** [comment]
@UnleashedMind I was already on 1.0.6, nevertheless I upgraded to 1.0.7 but it still redeploys the backend. If that makes any difference, it is a Python backend. Here are the logs:

```
ali@X230:~/ws-js/upload$ npm --version
5.6.0
ali@X230:~/ws-js/upload$ node --version
v6.11.4
ali@X230:~/ws-js/upload$ awsmobile --version

1.0.7

ali@X230:~/ws-js/upload$ git diff
ali@X230:~/ws-js/upload$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
ali@X230:~/ws-js/upload$ awsmobile pull

retrieving the latest backend awsmobile project information
awsmobile project's details logged at: awsmobilejs/#current-backend-info/backend-details.json
awsmobile project's access information logged at: awsmobilejs/#current-backend-info/aws-exports.js
awsmobile project's access information copied to: src/aws-exports.js
awsmobile project's specifications logged at: awsmobilejs/#current-backend-info/mobile-hub-project.yml
contents in #current-backend-info/ is synchronized with the latest in the aws cloud
? sync corresponding contents in backend/ with #current-backend-info/ No

ali@X230:~/ws-js/upload$ awsmobile publish 

building backend
   building cloud-api
      zipping resizeImage
   generating backend project content
   backend project content generation successful
done, build artifacts are saved at: 
/home/ali/ws-js/upload/awsmobilejs/.awsmobile/backend-build

preparing for backend project update: upload
   uploading resizeImage-20180109221107.zip
   upload Successful  resizeImage-20180109221107.zip
done

updating backend project: upload
awsmobile update api call returned with no error
waiting for the formation of cloud-api to complete
cloud-api update finished with status code: UPDATE_COMPLETE

Successfully updated the backend awsmobile project: upload

retrieving the latest backend awsmobile project information
awsmobile project's details logged at: awsmobilejs/#current-backend-info/backend-details.json
awsmobile project's access information logged at: awsmobilejs/#current-backend-info/aws-exports.js
awsmobile project's access information copied to: src/aws-exports.js
awsmobile project's specifications logged at: awsmobilejs/#current-backend-info/mobile-hub-project.yml
contents in #current-backend-info/ is synchronized with the latest in the aws cloud

executing npm install ...
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

up to date in 10.331s
npm install exited with code 0

executing frontend build command ...
```
(and then it rebuilds and redeploys the frontend too, omitting the logs)

**What should I do to help you find the problem?**

**Comment by UnleashedMind** [comment]
`awsmobile publish` is just a convenience command that internally invokes push before other actions, and the `awsmobile push` checks timestamps to avoid unnecessary actions. 
If you could please execute `awsmobile push` twice in a roll without modifying anything in between, see if the second push still calls mobile hub updateProject API method. It will provide us with more information.

**Comment by baharev** [comment]
@UnleashedMind The second push redeploys the backend again, here is the log:

```
ali@X230:~/ws-js/upload$ nvm use
Found '/home/ali/ws-js/upload/.nvmrc' with version <lts/boron>
Now using node v6.11.4 (npm v5.6.0)
ali@X230:~/ws-js/upload$ awsmobile --version

1.0.7

ali@X230:~/ws-js/upload$ awsmobile push

building backend
   building cloud-api
      zipping resizeImage
   generating backend project content
   backend project content generation successful
done, build artifacts are saved at: 
/home/ali/ws-js/upload/awsmobilejs/.awsmobile/backend-build

preparing for backend project update: upload
   uploading resizeImage-20180110222531.zip
   upload Successful  resizeImage-20180110222531.zip
done

updating backend project: upload
awsmobile update api call returned with no error
waiting for the formation of cloud-api to complete
cloud-api update finished with status code: UPDATE_COMPLETE

Successfully updated the backend awsmobile project: upload

retrieving the latest backend awsmobile project information
awsmobile project's details logged at: awsmobilejs/#current-backend-info/backend-details.json
awsmobile project's access information logged at: awsmobilejs/#current-backend-info/aws-exports.js
awsmobile project's access information copied to: src/aws-exports.js
awsmobile project's specifications logged at: awsmobilejs/#current-backend-info/mobile-hub-project.yml
contents in #current-backend-info/ is synchronized with the latest in the aws cloud

ali@X230:~/ws-js/upload$ awsmobile push

building backend
   building cloud-api
      zipping resizeImage
   generating backend project content
   backend project content generation successful
done, build artifacts are saved at: 
/home/ali/ws-js/upload/awsmobilejs/.awsmobile/backend-build

preparing for backend project update: upload
   uploading resizeImage-20180110223023.zip
   upload Successful  resizeImage-20180110223023.zip
done

updating backend project: upload
awsmobile update api call returned with no error
waiting for the formation of cloud-api to complete
cloud-api update finished with status code: UPDATE_COMPLETE

Successfully updated the backend awsmobile project: upload

retrieving the latest backend awsmobile project information
awsmobile project's details logged at: awsmobilejs/#current-backend-info/backend-details.json
awsmobile project's access information logged at: awsmobilejs/#current-backend-info/aws-exports.js
awsmobile project's access information copied to: src/aws-exports.js
awsmobile project's specifications logged at: awsmobilejs/#current-backend-info/mobile-hub-project.yml
contents in #current-backend-info/ is synchronized with the latest in the aws cloud

ali@X230:~/ws-js/upload$ 
```

I am happy help, just let me know how I can help you further.

**Comment by baharev** [comment]
If that is any help, I created the Python lambda function in the browser at https://eu-central-1.console.aws.amazon.com/lambda/home?region=eu-central-1#/functions 

This led to complications, the first pull hung forever, here is the log:

```
ali@X230:~/ws-js/upload$ awsmobile pull

retrieving the latest backend awsmobile project information
awsmobile project's details logged at: awsmobilejs/#current-backend-info/backend-details.json
awsmobile project's access information logged at: awsmobilejs/#current-backend-info/aws-exports.js
awsmobile project's access information copied to: src/aws-exports.js
awsmobile project's specifications logged at: awsmobilejs/#current-backend-info/mobile-hub-project.yml
⠹ downloading cloud-api lambda function codebases{ AccessDeniedException: User: arn:aws:iam::123456789012:user/AWSMobileCLI is not authorized to perform: lambda:GetFunction on resource: arn:aws:lambda:eu-central-1:123456789012:function:resizeImage
    at Object.extractError (/home/ali/.nvm/versions/node/v6.11.4/lib/node_modules/awsmobile-cli/node_modules/aws-sdk/lib/protocol/json.js:48:27)
    at Request.extractError (/home/ali/.nvm/versions/node/v6.11.4/lib/node_modules/awsmobile-cli/node_modules/aws-sdk/lib/protocol/rest_json.js:52:8)
    at Request.callListeners (/home/ali/.nvm/versions/node/v6.11.4/lib/node_modules/awsmobile-cli/node_modules/aws-sdk/lib/sequential_executor.js:105:20)
    at Request.emit (/home/ali/.nvm/versions/node/v6.11.4/lib/node_modules/awsmobile-cli/node_modules/aws-sdk/lib/sequential_executor.js:77:10)
    at Request.emit (/home/ali/.nvm/versions/node/v6.11.4/lib/node_modules/awsmobile-cli/node_modules/aws-sdk/lib/request.js:683:14)
    at Request.transition (/home/ali/.nvm/versions/node/v6.11.4/lib/node_modules/awsmobile-cli/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/home/ali/.nvm/versions/node/v6.11.4/lib/node_modules/awsmobile-cli/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /home/ali/.nvm/versions/node/v6.11.4/lib/node_modules/awsmobile-cli/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/home/ali/.nvm/versions/node/v6.11.4/lib/node_modules/awsmobile-cli/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/home/ali/.nvm/versions/node/v6.11.4/lib/node_modules/awsmobile-cli/node_modules/aws-sdk/lib/request.js:685:12)
  message: 'User: arn:aws:iam::123456789012:user/AWSMobileCLI is not authorized to perform: lambda:GetFunction on resource: arn:aws:lambda:eu-central-1:123456789012:function:resizeImage',
  code: 'AccessDeniedException',
  time: 2017-12-31T13:14:42.290Z,
  requestId: '90270f3b-ee2c-11e7-92e4-239b59d67d10',
  statusCode: 403,
  retryable: false,
  retryDelay: 89.04204910440667 }
⠴ downloading cloud-api lambda function codebases

```

(I killed this command after a really long while)

Then I attached this to the awsmobile cli:

```
{
    "Version": "2012-10-17",
    "Statement": {
        "Effect": "Allow",
        "Action": "lambda:GetFunction",
        "Resource": "*"
    }
}
```

Although I wanted to open a new issue, ask you to fix that as well, and generate and attach that policy automagically too.

**Comment by baharev** [comment]
@UnleashedMind Just pinging: Can I do something to accelerate things?

**Comment by UnleashedMind** [comment]
We are still investigating this issue, we will notify you when there is an update.

**Comment by UnleashedMind** [comment]
The latest cli update has refactored the push timestamp check logic, please check to see if it resolves this issue. Thanks

**Comment by baharev** [comment]
**No, unfortunately v1.0.12 does not solve the problem.**

If I do two `awsmobile publish` in a row, immediately one after the other, it still redeploys the backend (and the frontend too :( ). See the log below.

Note that as mentioned in #45 the zip file of a single publish appears 3-5 times in S3. See a screenshot below. That screenshot was made two weeks ago but awsmobile v1.0.12 does the same today. I double-checked.

Also note that the triggers for my Lambda have been mysteriously duplicated (see the screenshot below). I have 2 triggers, and each one has been duplicated somehow...

-----

```
ali@X230:~/ws-js/upload$ awsmobile --version

1.0.12

ali@X230:~/ws-js/upload$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
ali@X230:~/ws-js/upload$ git diff
ali@X230:~/ws-js/upload$ awsmobile pull

retrieving the latest backend awsmobile project information
awsmobile project's details logged at: awsmobilejs/#current-backend-info/backend-details.json
awsmobile project's access information logged at: awsmobilejs/#current-backend-info/aws-exports.js
awsmobile project's access information copied to: src/aws-exports.js
awsmobile project's specifications logged at: awsmobilejs/#current-backend-info/mobile-hub-project.yml
contents in #current-backend-info/ is synchronized with the latest in the aws cloud
? sync corresponding contents in backend/ with #current-backend-info/ Yes

ali@X230:~/ws-js/upload$ git diff
ali@X230:~/ws-js/upload$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
ali@X230:~/ws-js/upload$ awsmobile publish

checking for local backend changes
   building cloud-api
      zipping resizeImage
   done
   generating backend project content
   done
backend build artifacts are saved at: 
/home/ali/ws-js/upload/awsmobilejs/.awsmobile/backend-build

preparing for backend project update: upload
   uploading resizeImage-20180203142651.zip
   upload Successful  resizeImage-20180203142651.zip
done

updating backend project: upload
awsmobile update api call returned with no error
waiting for the cloud formation operations to complete
cloud-api update finished with status code: UPDATE_COMPLETE

Successfully updated the backend awsmobile project: upload

retrieving the latest backend awsmobile project information
awsmobile project's details logged at: awsmobilejs/#current-backend-info/backend-details.json
awsmobile project's access information logged at: awsmobilejs/#current-backend-info/aws-exports.js
awsmobile project's access information copied to: src/aws-exports.js
awsmobile project's specifications logged at: awsmobilejs/#current-backend-info/mobile-hub-project.yml
contents in #current-backend-info/ is synchronized with the latest in the aws cloud

executing npm install ...
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

up to date in 10.551s
npm install exited with code 0

executing frontend build command ...

> upload@0.1.0 build /home/ali/ws-js/upload
> react-scripts build

Creating an optimized production build...
Compiled successfully.

File sizes after gzip:

  210.68 KB (+1.26 KB)  build/static/js/main.c43f0883.js
  362 B (+24 B)         build/static/css/main.1f1b0871.css

The project was built assuming it is hosted at ./.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.

build command exited with code 0

Successful!
your application is published and hosted at:
http://upload-hosting-mobilehub-123456789.s3-website.eu-central-1.amazonaws.com
your application is also distributed through aws CloudFront
https://abcdefghijklm.cloudfront.net
please note that CloudFront content may not have been refreshed

ali@X230:~/ws-js/upload$ awsmobile publish

checking for local backend changes
   building cloud-api
      zipping resizeImage
   done
   generating backend project content
   done
backend build artifacts are saved at: 
/home/ali/ws-js/upload/awsmobilejs/.awsmobile/backend-build

preparing for backend project update: upload
   uploading resizeImage-20180203143115.zip
   upload Successful  resizeImage-20180203143115.zip
done

updating backend project: upload
awsmobile update api call returned with no error
waiting for the cloud formation operations to complete
cloud-api update finished with status code: UPDATE_COMPLETE

Successfully updated the backend awsmobile project: upload

retrieving the latest backend awsmobile project information
awsmobile project's details logged at: awsmobilejs/#current-backend-info/backend-details.json
awsmobile project's access information logged at: awsmobilejs/#current-backend-info/aws-exports.js
awsmobile project's access information copied to: src/aws-exports.js
awsmobile project's specifications logged at: awsmobilejs/#current-backend-info/mobile-hub-project.yml
contents in #current-backend-info/ is synchronized with the latest in the aws cloud

executing npm install ...
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

up to date in 10.726s
npm install exited with code 0

executing frontend build command ...

> upload@0.1.0 build /home/ali/ws-js/upload
> react-scripts build

Creating an optimized production build...
Compiled successfully.

File sizes after gzip:

  210.68 KB  build/static/js/main.c43f0883.js
  362 B      build/static/css/main.1f1b0871.css

The project was built assuming it is hosted at ./.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.

build command exited with code 0

Successful!
your application is published and hosted at:
http://upload-hosting-mobilehub-123456789.s3-website.eu-central-1.amazonaws.com
your application is also distributed through aws CloudFront
https://abcdefghijklm.cloudfront.net
please note that CloudFront content may not have been refreshed

ali@X230:~/ws-js/upload$ 
```

----

**Each publish seem to create the same zip file 3-5 times (even with awsmobile v1.0.12)**

![why_that_many_versions](https://user-images.githubusercontent.com/2605584/35767867-d5a87cb0-08f3-11e8-8303-8d14dd96dd96.png)

----

**Triggers for Lambda somehow duplicated**

![screenshot_2018-01-28_01-29-48](https://user-images.githubusercontent.com/2605584/35767843-757e5ac6-08f3-11e8-9738-1d42e892f456.png)

**Comment by UnleashedMind** [comment]
There are more than one issues here
Issue 1: unnecessary backend push
  I found this line suspicious `The project was built assuming it is hosted at ./.'  
  does your frontend build command modify the awsmobilejs/backend folder or its contents? 
  more precisely, does the frontend build command change the 'mtime' of the awsmobile/backend
folder or its contents? 
  If it does, then the cli will have to build and push again, because the cli checks the mtime and concludes the backend has been modified since last push. 
  Please share your frontend build command scripts.

Issue 2, It's a server side issue, we've contacted the mobile hub server side team and it's in their backlog to work on that. To follow the "separation of concern" principle, let's use issue #45 to track it.

**Comment by UnleashedMind** [comment]
Please executed `awsmobile configure project -l` on your project and paste the result here. 

The awsmobile-cli provides the convenience of developing the backend and frontend codebases in a single project, but it needs a clear separation of the frontend and backend codebases, including the build/dist folders for it to work correctly. Are there any overlaps in those folders?

**Comment by baharev** [comment]
I have created the frontend with `create-react-app`, and the the Python lambda function in the browser at https://eu-central-1.console.aws.amazon.com/lambda/home?region=eu-central-1#/functions 

Everything was left at its default value; I did not mess with default directories / paths.

> does your frontend build command modify the awsmobilejs/backend folder or its contents?
more precisely, does the frontend build command change the 'mtime' of the awsmobile/backend
folder or its contents?

No, not to my knowledge, but I will try to figure it out.

> Issue 2, It's a server side issue, we've contacted the mobile hub server side team and it's in their backlog to work on that. To follow the "separation of concern" principle, let's use issue #45 to track it.

OK, understood.

> The awsmobile-cli provides the convenience of developing the backend and frontend codebases in a single project, but it needs a clear separation of the frontend and backend codebases, including the build/dist folders, for it to work correctly. Are there any overlaps in those folders?

No, not to my knowledge. I left everything at its default value. But now I know what to check!

If there is an overlap, then the default paths must be changed because other users will run into the same issue too.

> Please share your frontend build command scripts.
> Please execute `awsmobile configure project -l` on your project and paste the result here.

Coming right up.

**Comment by baharev** [comment]
> I found this line suspicious `The project was built assuming it is hosted at ./.'

That comes from `package.json`, in particular this line:

```
    "homepage": ".",
````

and that is considered normal (as far as I know). See also #6 that is concerned with this attribute.

> does your frontend build command modify the awsmobilejs/backend folder or its contents?

No, it shouldn't. As far as I can tell, it is completely unaware of the existence of that directory, and it really shouldn't mess with it.

> more precisely, does the frontend build command change the 'mtime' of the awsmobile/backend
folder or its contents?

No. The frontend does not know that that directory exists, and it has no interest in reaching into that. I also see absolutely no reason why it would do so: One is a react frontend in JavaScript, the other is a Python lambda function for the backend. They have no knowledge of each other.

> Please share your frontend build command scripts.

They are under: `<my project>/node_modules/react-scripts/`. I have attached the whole directory as a zip file "as is": [react-scripts.zip](https://github.com/aws/awsmobile-cli/files/1692216/react-scripts.zip)

You will also need the my `package.json`, here: [package.json.zip](https://github.com/aws/awsmobile-cli/files/1692217/package.json.zip)

> Please execute awsmobile configure project -l on your project and paste the result here.
```
ali@X230:~/ws-js/upload$ awsmobile configure project -l

{ SourceDir: 'src',
  DistributionDir: 'build',
  BuildCommand: 'npm run-script build',
  StartCommand: 'npm run-script start' }
```

By the way, why does `awsmobile publish` run `npm install`? An `npm build` would suffice...

> The awsmobile-cli provides the convenience of developing the backend and frontend codebases in a single project, but it needs a clear separation of the frontend and backend codebases, including the build/dist folders, for it to work correctly. Are there any overlaps in those folders?

No. 

And as I argued above, they have no interest in reaching into each others code / directories.

Please let me know if you need further help.

**Comment by UnleashedMind** [comment]
Thank you very much for your patience, this is not typical and I am having a hard time trying to reproduce it. 
What might be helpful is that if you could let us know how your lambda function is setup and how it is integrated into the cloud-api feature of your backend project. That might give us clue on what might have caused this annoying unnecessary backend update.
Also, please locate your project's awsmobilejs/.awsmobile/info/project-info.json file, and let us know its contents after a successful awsmobile push. That file records the operation timestamps that are used by the cli.

**Comment by baharev** [comment]
> What might be helpful is that if you could let us know how your lambda function is setup and how it is integrated into the cloud-api feature of your backend project. That might give us clue on what might have caused this annoying unnecessary backend update.

Please read my [previous comment above](https://github.com/aws/awsmobile-cli/issues/34#issuecomment-356749092), including the issues that occurred on the first `awsmobile pull` and how I resolved that.

Please be more specific, and tell me what information you need (which files, what command to run, etc).

> Also, please locate your project's awsmobilejs/.awsmobile/info/project-info.json file, and let us know its contents after a successful awsmobile push. That file records the operation timestamps that are used by the cli.

I did two `awsmobile publish` in a row. After the first one, I haven't touched anything, and I haven't done anything; I immediately issued the second `awsmobile publish`. Here are the requested files after the first, then after the second `awsmobile publish`. 

[after-first-project-info.json.zip](https://github.com/aws/awsmobile-cli/files/1692315/after-first-project-info.json.zip)

[after-second-project-info.json.zip](https://github.com/aws/awsmobile-cli/files/1692316/after-second-project-info.json.zip)

Note: I removed my project's UUID. (I don't know what sensitive data is here, and what isn't, so just to be on the safe side.)

The second publish shouldn't have done anything in my opinion: The frontend hasn't changed either, but it even ran `npm install`. :( I don't know where `npm install` is coming from; `npm build` would suffice even if the frontend had been changed.

In other words: It is not just the backend that is being unnecessarily redeployed.

**Comment by UnleashedMind** [comment]
The update check has been completely refactored. 
Closing this issue because it does not reflect the behavior of the latest cli version (1.1.x and above). 
Feel free to create a new issues.

**Comment by baharev** [comment]
@UnleashedMind OK, I will check it during the weekend.

**Comment by nawwa** [comment]
Any news @baharev , I might have the same issue

**Comment by baharev** [comment]
@nawwa Ooops, sorry, I haven't checked this one since. I will try to get back to you this evening but I cannot make any promises.

**Comment by nawwa** [comment]
@baharev No problem man :) !

**Comment by kunwarVivek** [comment]
This problem still exists. Just create default awsmobile template using "awsmobile start testapp react" command. Then publish the app using "awsmobile publish" twice. 
The command takes a lot of time and even though nothing has changed , does
- build and push the backend code
- the cloudformation creation again
- the frontend build and upload to S3 bucket
The awsmobile cli version is 1.1.5 and the above tests are done on windows 10.

**Comment by baharev** [comment]
I ended up writing my own script for deployment. Then I know what happens, no magic. If there is an issue, I can fix it too, and I don't have to wait several months. I am not using awsmobile (or the newer amplify-cli) anymore.

**Comment by nawwa** [comment]
@baharev Do your script push the awsmobile backend ? Im interested on how you did this :)

**Comment by baharev** [comment]
@nawwa Hmmm, I am not quite sure what you mean by that. I wrote my own Python script that copies the necessary files to the S3 bucket. I did not need more.

**Comment by nawwa** [comment]
@baharev Oh ok, nice ! I was speaking of the lambdas filess

**Comment by baharev** [comment]
@nawwa OK, I see. I haven't tried it myself yet, but that should not be significantly more complicated, and I am sure it is well documented too (the Lambda REST API).

**Comment by nawwa** [comment]
@baharev Yeah :D

**Comment by baharev** [comment]
@nawwa OK, I take it as a no... What would you need on the backend side? If I need it to, I will check it. I only need Lamda. You?

**Comment by nawwa** [comment]
(Sorry for the late answer)
No, no its was a true "Yes", I agree with you, I'm sure the doc provide everything.
I needed the lambdas code to be pushed on the AWS account without changing the lambdas configuration. What I mean by "configuration" is VPC access for example. Everytime I push my lambdas using awsmobile client, my VPC configuration is removed 

Don't bother lookin' into it btw, I answered just because you asked ! 👍

**Comment by baharev** [comment]
@nawwa OK, I understand. I mostly moved away from awsmobile and the newer Amplify CLI. I still use them to see what configurations are necessary, but once I know what to do, I write my own Python script for deployment, and then I have full control.

**Comment by nawwa** [comment]
@baharev Yeah it seems like this is the most clever way to work. Thanks for your answers.
